File: taskCloudService/src/repo_match_key_test.go
package main

import "testing"

func TestRepoMatchKeyFromURL(t *testing.T) {
	cases := []struct {
		in, want string
	}{
		{"https://github.com/acme/demo.git", "github.com/acme/demo"},
		{"http://localhost:8012/ljy/somanyad.git", "localhost:8012/ljy/somanyad"},
		{"https://gitlab.daydaymoney.com/example-user/somanyad-emailD.git", "gitlab.daydaymoney.com/example-user/somanyad-emaild"},
		{"git@github.com:Acme/Demo.git", "github.com/acme/demo"},
		{"ssh://git@github.com/acme/demo.git", "github.com/acme/demo"},
		{"ssh://git@gitlab.daydaymoney.com:2222/g/p.git", "gitlab.daydaymoney.com/g/p"},
		{"", ""},
	}
	for _, tc := range cases {
		got := repoMatchKeyFromURL(tc.in)
		if got != tc.want {
			t.Errorf("repoMatchKeyFromURL(%q)=%q want %q", tc.in, got, tc.want)
		}
	}
}

File: trae-agent/onlineServiceJS/src/repoMatchKey.test.mjs
import assert from 'node:assert/strict';
import test from 'node:test';

import { canonicalRepoKey, repoMatchKeyFromUrl } from './repoMatchKey.mjs';
import { repoMatchKeyFromUrl as repoMatchKeyFromHelpers } from './layerGitRouteHelpers.mjs';

test('repoMatchKeyFromUrl strips scheme and .git', () => {
  assert.equal(
    repoMatchKeyFromUrl('https://gitlab-tencent-sh-1.daydaymoney.com/example-user/somanyad.git'),
    'gitlab-tencent-sh-1.daydaymoney.com/example-user/somanyad',
  );
});

test('repoMatchKeyFromUrl equates ssh:// and git@ without ssh port', () => {
  assert.equal(
    repoMatchKeyFromUrl('ssh://git@github.com/acme/demo.git'),
    'github.com/acme/demo',
  );
  assert.equal(
    repoMatchKeyFromUrl('ssh://git@gitlab.daydaymoney.com:2222/g/p.git'),
    repoMatchKeyFromUrl('git@gitlab.daydaymoney.com:g/p.git'),
  );
});


test('layerGitRouteHelpers re-exports repoMatchKeyFromUrl', () => {
  assert.equal(
    repoMatchKeyFromHelpers('ssh://git@gitlab.daydaymoney.com:2222/g/p.git'),
    repoMatchKeyFromUrl('git@gitlab.daydaymoney.com:g/p.git'),
  );
});

test('canonicalRepoKey keeps scheme and lowercases', () => {
  assert.equal(
    canonicalRepoKey('https://gitlab-tencent-sh-1.daydaymoney.com/example-user/somanyad.git'),
    'https://gitlab-tencent-sh-1.daydaymoney.com/example-user/somanyad',
  );
});
